### PR TITLE
Replace flake8 with Ruff

### LINKED ISSUE #586: Replace flake8 with Ruff
https://docs.astral.sh/ruff/

### PR DESCRIPTION
**Pull request by am009**
To replace flake8 with Ruff, there are roughly two steps:

- Write a ruff config corresponding to the original `.flake8` file.
- Update the precommit hook to use flake8. (According to [here](https://docs.astral.sh/ruff/tutorial/#integrations))

Some notes:

- W503 (line break before binary operator) and B950 (line too long, but [a bit weird](https://github.com/PyCQA/flake8-bugbear/issues/54)) seem to be not implemented in ruff. But they are also (mostly) disabled in previous flake8 config file.

fixes #586

Checklist:

- [ ] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [ ] Add or update relevant docs, in the `docs` folder and in code docstring.
- [ ] Add an entry in `CHANGES.md` summarizing the change and linking to the issue.
- [ ] Add `*Version changed*` or `*Version added*` note in any relevant docs and docstring.
- [x] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,16 +16,12 @@ repos:
...
     hooks:
-      - id: flake8
-        additional_dependencies:
-          - flake8-bugbear
-          - flake8-implicit-str-concat
-          - flake8-quotes
-        args:
-          - --config=.flake8
+      # Run the linter.
+      - id: ruff
```
Comment (greyli) [inline .pre-commit-config.yaml:24]: We could also enable the formatter:
```
- id: ruff-format
```

File: pyproject.toml [hunk 2]
```diff
@@ -76,3 +76,44 @@ addopts = ["--cov=src/apiflask", "--cov-branch", "--cov-report=term-missing"]
...
+line-length = 100
+exclude = [
+    ".*",
+    "*.txt",
+    "*.md",
+    "__pycache__",
+]
+
+[tool.ruff.format]
+# Like Black, use double quotes for strings.
```
Comment (greyli) [inline pyproject.toml:91]: This comment is outdated and should be removed.

**Review by greyli — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (am009) [reply .pre-commit-config.yaml:24]: Do we really need to format the code? After adding this and run `pre-commit run --all-files`, there are tons of changes:

image

Should I commit all these changes? Or there is some way to handle this but I don't know

Comment (greyli) [reply .pre-commit-config.yaml:24]: Yes, commit all the changes. It's a coding style refresh for the code base.

**Comment by am009** [comment]
I have updated my code and rebased to the latest commit on main branch.

**Review by greyli — APPROVED** [review]

**Comment by greyli** [comment]
Merged, thanks!

**Pull request merged**